sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVOptions.scala

```diff
@@ -90,6 +90,7 @@ class CSVOptions(
   val quote = getChar("quote", '\"')
   val escape = getChar("escape", '\\')
   val comment = getChar("comment", '\u0000')
+  val lineSeparator = parameters.getOrElse("lineSeparator", "\n")
```

Comment:
To cut this short, up to my knowledge, for the current status (please correct me if I am wrong),

CSV read -> cover `\n` and `\r\n`
CSV read with `wholeFile` (`multiLine`) -> OS dependent newline (by Univocity)
CSV write -> OS dependent newline (by Univocity)

JSON read -> cover `\n` and `\r\n`
JSON read with `wholeFile` (`multiLine`) -> N/A (it reads the whole file as a single record for the current status)
JSON write -> `\n`

TEXT read -> cover `\n` and `\r\n`
TEXT write -> `\n`

For hardcorded newline, fixing it to `\n` is probably better. I wouldn't mind if this PR is turned to fix this for 'CSV read with wholeFile (multiLine)' and 'CSV write'.

For configurable line separator, it would be not configurable in this way for 'CSV read'. We should be able to read it back what we write out.